includes/User.php

```diff
@@ -4953,6 +4915,46 @@ protected function loadOptions() {
 		wfRunHooks( 'UserLoadOptions', array( $this, &$this->mOptions ) );
 	}
 
+	private function loadAttributes() {
+		global $wgEnableReadsFromAttributeService;
+
+		if ( !empty( $wgEnableReadsFromAttributeService ) ) {
+			$attributes = $this->userAttributes()->getAttributes($this->getId());
```

Comment:
There is a couple of $this->userAttributes()->[something] calls.
Is there a way or would it be possible to cache the userAttributes object without calling the Injector each time?